MediaBundle/Twig/DisplayMediaExtension.php

```diff
@@ -106,6 +107,22 @@ public function getMediaAlt($mediaId)
     }
 
     /**
+     * @param string $mediaId
+     *
+     * @return string
+     */
+    public function getMediaTitle($mediaId)
+    {
+        $media = $this->getMedia($mediaId);
+
+        if ($media) {
+            return $media->getTitle($this->request->getLocale());
```

Comment:
You should not use the request->getLocale() but the request->get('language', $request->getLocale()) parameter